src/components/QuickProjectSearch.tsx

```diff
@@ -0,0 +1,171 @@
+import { ArrowRightOutlined, CloseCircleOutlined } from '@ant-design/icons'
+import { t } from '@lingui/macro'
+import Input from 'antd/lib/input/Input'
+import Modal from 'antd/lib/modal/Modal'
+import { PV_V2 } from 'constants/pv'
+import { useProjectsSearch } from 'hooks/Projects'
+import { useRouter } from 'next/router'
+import React, { useCallback, useEffect, useState } from 'react'
+import { twMerge } from 'tailwind-merge'
+
+import Loading from './Loading'
+import { ProjectVersionBadge } from './ProjectVersionBadge'
+import V1ProjectHandle from './v1/shared/V1ProjectHandle'
+import V2V3ProjectHandleLink from './v2v3/shared/V2V3ProjectHandleLink'
+
+const HOT_KEY = 'k'
+const INPUT_ID = 'quickProjectSearch'
+const MAX_RESULTS = 8
+
+export default function QuickProjectSearch() {
+  const [searchText, setSearchText] = useState<string>()
+  const [highlightIndex, setHighlightIndex] = useState<number>()
+  const [spotlightActive, setSpotlightActive] = useState<boolean>()
+
+  const router = useRouter()
+
+  const { data: searchPages, isLoading: isLoadingSearch } =
+    useProjectsSearch(searchText)
+
+  const goToProject = useCallback(() => {
+    if (highlightIndex === undefined) return
+
+    const project = searchPages?.[highlightIndex]
+
+    if (!project) return
+
+    router.push(
+      project.pv === PV_V2 ? `/@${project.handle}` : `/p/${project.handle}`,
+    )
+  }, [router, searchPages, highlightIndex])
+
+  const reset = useCallback(() => {
+    setSpotlightActive(false)
+    setHighlightIndex(undefined)
+    setSearchText(undefined)
+  }, [])
+
+  // Hot key listener
+  useEffect(() => {
+    const listener = (e: KeyboardEvent) => {
+      if (e.isComposing) return // i think right idk
+
+      if (spotlightActive) {
+        switch (e.key) {
+          case 'escape':
+            reset()
+            break
+          case 'Return':
+          case 'Enter':
+            goToProject()
+            reset()
+            break
+        }
+      } else if (e.key === HOT_KEY && (e.metaKey || e.ctrlKey)) {
+        setSpotlightActive(true)
+        setTimeout(
```

Comment:
NBD but I think we can use refs to solve this - I read about them a month or so ago and this is their exact use case, but lets worry about it when its a problem (may never be)

Reply:
good suggestion. i'm about to add a debounce to the typing to limit graph requests, and i'll look at this in the same PR. gonna merge this for now since everything seems functional at least